proposed/2023/project-exclusion-for-transitive-dependencies.md

```diff
@@ -0,0 +1,130 @@
+# Support excluding projects from transitive dependencies
+
+- Author Name [Greybird](https://github.com/Greybird)
+- Start Date 2023-07-11
+- GitHub Issue [12585](https://github.com/NuGet/Home/issues/12585)
+- GitHub PR https://github.com/NuGet/NuGet.Client/pull/5302
+
+## Summary
+
+Currently, when listing transitive packages using `dotnet list package --include-transitive`, both project and packages dependencies are reported. The goal of this proposal is to provide a way to filter the projects out of the report.
+
+## Motivation 
+
+Packages and projects are two vastly different types of references.
+Packages are published to a repository, allowing consumption by multiple pieces of software and repositories. Metadata are attached to packages or package versions, either included in the package, or added as a decoration - for deprecation or vulnerabilities for example -, which can be consumed to take decisions. On the contrary, projects are _local_ entities, living in a single repository, and do not have an existence outside of it.
+
+The `list package` command already supports filtering the results to retrieve only deprecated (`--deprecated`), vulnerable (`--vulnerable`), or not up-to-date (`--outdated`) packages. These commands only return actual packages, which is understandable as the underlying information used to choose what to report is managed by the _package_ manager, aka nuget.
+
+Allowing to remove projects from the output when none of these flags is provided would allow users to get an accurate view of their package dependencies. This would ensure that the output of the command can be processed in an uniform way by either a user, or a piece of software - through the machine readable output - to support any report, or process associated with packages, even those with no issues related to deprecation, vulnerabilities, or up-to-dateness.
```

Comment:
Testing on the latest .NET SDK, projects *are* not included in the default output at all. 
With `--include-transitive`, they are included in the transitive packages (which is incorrect). 

See my example from above: 

Project Root -> Project INtermediate (1.0.0) -> Project Leaf (1.0.0)
Project Root -> Package Leaf (2.0.0)

In this case the output is:

```xml
Project 'Root' has the following package references
   [net8.0]:
   Top-level Package      Requested   Resolved
   > Leaf                 2.0.0       2.0.0

   Transitive Package      Resolved
   > Intermediate          1.0.0
```

Seems like there's some intent behind that and we migt just be dealing with a bug.

Reply:
I added details on the behavior of the command, and mentioned that packages are reported as transitive dependencies hile they are direct, for the sake of clarity.